### PR TITLE
Issue #47: Save events room database

### PR DESCRIPTION
**Pull request by SalosV**
**Save events room database** #47 
- The event table was create.
- The counter query, query and insertion of event data was created

**Testing Done:**

- Huawei p20 Lite

**How to test:**

- Run the application and click in events
- You can see the loading while the event have been loaded
- Then you can see the events in a scrollable list
- close the application 
- Disable internet
- Open the application and click in events

**Signed-off-by**:

- Santiago Lozada <santiagolozada95@gmail.com>

### CODE REVIEW

File: data/src/main/java/com/architect/coders/mu8/data/events/EventsEntity.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.architect.coders.mu8.data.events
+
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+import com.architect.codes.mu8.events.Event
+import com.architect.codes.mu8.utils.EMPTY_LONG
+import com.architect.codes.mu8.utils.EMPTY_STRING
+
```
Comment (aliceresponde) [inline data/src/main/java/com/architect/coders/mu8/data/events/EventsEntity.kt:8]: check if do you need select a specific event.

File: data/src/main/java/com/architect/coders/mu8/data/events/EventsRepositoryImpl.kt [hunk 2]
```diff
@@ -1,30 +1,40 @@
...
+    private val database = application.database
 
-        val events = mutableListOf<Event>()
-        if (response.isSuccessful) {
-            response.body()?.data?.results?.forEach {
-                events.add(mapper.transform(it))
+    override suspend fun invoke(): List<Event> = withContext(Dispatchers.IO) {
+
+        with(database.getEventsDao()) {
+
```
Comment (aliceresponde) [inline data/src/main/java/com/architect/coders/mu8/data/events/EventsRepositoryImpl.kt]: remove line

File: data/src/main/java/com/architect/coders/mu8/data/events/EventsEntity.kt (hunk 1 above)
Comment (SalosV) [reply data/src/main/java/com/architect/coders/mu8/data/events/EventsEntity.kt:8]: Done

File: data/src/main/java/com/architect/coders/mu8/data/events/EventsRepositoryImpl.kt (hunk 2 above)
Comment (SalosV) [reply data/src/main/java/com/architect/coders/mu8/data/events/EventsRepositoryImpl.kt]: Done

**Review by recaldev — APPROVED** [review]

**Pull request merged**